cli/preprocessor/transaction.go

```diff
@@ -0,0 +1,87 @@
+package preprocessor
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/goccy/go-yaml"
+	"github.com/kubeshop/tracetest/cli/openapi"
+	"github.com/kubeshop/tracetest/cli/pkg/fileutil"
+	"go.uber.org/zap"
+)
+
+type applyTestFunc func(context.Context, fileutil.File) (fileutil.File, error)
+
+type transaction struct {
+	logger      *zap.Logger
+	applyTestFn applyTestFunc
+}
+
+func Transaction(logger *zap.Logger, applyTestFn applyTestFunc) transaction {
+	return transaction{
+		logger:      logger,
+		applyTestFn: applyTestFn,
+	}
+}
+
+func (t transaction) Preprocess(ctx context.Context, input fileutil.File) (fileutil.File, error) {
+	var tran openapi.TransactionResource
```

Comment:
this is the same code moved from the `test_run_action.go` file, with a few variables renamed